### ISSUE TITLE
Chapter2的代码中编码器和解码器LN层是不是定义错了？

### ISSUE THREAD

**Issue by nors1993** [datawhalechina/happy-llm#47]
### 1. 遇到问题的章节 / Affected Chapter

https://github.com/datawhalechina/happy-llm/blob/main/docs/chapter2/code/transformer.py

### 2. 具体问题描述 / Problem Description

transformer架构里，不都是注意力层和FNN层后面各一个LN吗？这里为什么定义的跟架构里的位置不一致？
![Image](https://github.com/user-attachments/assets/0344ced9-a532-4fc7-957e-9ddaf4fa6ec5)

![Image](https://github.com/user-attachments/assets/f9a52bcd-a007-4b55-b350-b2a5effd8b46)

### 3. 问题重现材料 / Reproduction Materials

class EncoderLayer(nn.Module):
    def __init__(self, args):
        super().__init__()
        # 一个 Layer 中有两个 LayerNorm，分别在 Attention 之前和 MLP 之前
        self.attention_norm = LayerNorm(args.n_embd)
        # Encoder 不需要掩码，传入 is_causal=False
        self.attention = MultiHeadAttention(args, is_causal=False)
        self.fnn_norm = LayerNorm(args.n_embd)
        self.feed_forward = MLP(args.dim, args.dim, args.dropout)

class DecoderLayer(nn.Module):
    '''Decoder 层'''
    def __init__(self, args):
        super().__init__()
        # 一个 Layer 中有三个 LayerNorm，分别在 Mask Attention 之前、Self Attention 之前和 MLP 之前
        self.attention_norm_1 = LayerNorm(args.n_embd)
        # Decoder 的第一个部分是 Mask Attention，传入 is_causal=True
        self.mask_attention = MultiHeadAttention(args, is_causal=True)
        self.attention_norm_2 = LayerNorm(args.n_embd)
        # Decoder 的第二个部分是 类似于 Encoder 的 Attention，传入 is_causal=False
        self.attention = MultiHeadAttention(args, is_causal=False)
        self.ffn_norm = LayerNorm(args.n_embd)
        # 第三个部分是 MLP
        self.feed_forward = MLP(args.dim, args.dim, args.dropout)

### 确认事项 / Verification

- [x] 此问题未在过往Issue中被报告过 / This issue hasn't been reported before

**Comment by logan-zou** [comment]
已加入说明：

![Image](https://github.com/user-attachments/assets/b522b8a9-22c0-4c81-bf6c-05d1f0b7fc07)
